src/fhir/conditions.ts

```diff
@@ -174,3 +177,70 @@ function filterAndSort(conditions: ConditionModel[]): ConditionModel[] {
     ["desc"]
   );
 }
+
+export type SourceDocumentMap = Map<string, BinaryDocumentData>;
+
+export type BinaryDocumentData = {
+  xmlBinary: fhir4.Binary | undefined;
+  isBinary: boolean;
+};
+
+export async function getBinary(
+  requestContext: CTWRequestContext,
+  conditionsData?: CollapsibleDataListProps[]
+): Promise<SourceDocumentMap> {
+  // Call to Provenance for the conditionID to see if Binary exists. If it exists, use it to obtain the binary document and return that.
+
+  const binaryObjects: Map<string, BinaryDocumentData> = new Map();
+
+  // No conditions in the history drawer.
+  if (!conditionsData) {
+    return binaryObjects;
+  }
+
+  await Promise.all(
+    conditionsData.map(async (condition) => {
+      const endpointConditionsUrl = `${getZusApiBaseUrl(
+        requestContext.env
+      )}/fhir/Provenance?target=Condition/${condition.id}`;
+      const bundle = await fetch(endpointConditionsUrl, {
+        headers: {
+          Authorization: `Bearer ${requestContext.authToken}`,
+        },
+      });
+      const conditionsJSON = await bundle.json();
+
+      // The role should be of source otherwise can't be trusted to be provide the correct and truthy binary.
+      let documentBinary: fhir4.Binary | undefined;
+      if (conditionsJSON.entry) {
+        if (conditionsJSON.entry[0].resource.entity) {
+          if (conditionsJSON.entry[0].resource.entity[0].role === "source") {
+            const binaryID =
+              conditionsJSON.entry[0].resource.entity[0].what.reference;
+
+            const endpointBinaryUrl = `${getZusApiBaseUrl(
+              requestContext.env
+            )}/fhir/${binaryID}`;
+
+            const response = await fetch(endpointBinaryUrl, {
+              headers: {
+                Authorization: `Bearer ${requestContext.authToken}`,
+                accept: "application/json",
+                "Content-Type": "application/json+fhir",
+              },
+            });
+            documentBinary = await response.json();
+          }
+        }
+      }
```

Comment:
I am pretty sure we can change this to....

```suggestion
 if (conditionsJSON.entry[0].resource.entity[0].role === "source") {
        const binaryID =
          conditionsJSON.entry[0].resource.entity[0].what.reference;

        const endpointBinaryUrl = `${getZusApiBaseUrl(
          requestContext.env
        )}/fhir/${binaryID}`;

        const response = await fetch(endpointBinaryUrl, {
          headers: {
            Authorization: `Bearer ${requestContext.authToken}`,
            accept: "application/json",
            "Content-Type": "application/json+fhir",
          },
        });
        documentBinary = await response.json();
      }
```